doc/source/whatsnew/v2.0.0.rst

```diff
@@ -700,6 +700,7 @@ MultiIndex
 - Bug in :meth:`MultiIndex.union` not sorting when sort=None and index contains missing values (:issue:`49010`)
 - Bug in :meth:`MultiIndex.append` not checking names for equality (:issue:`48288`)
 - Bug in :meth:`MultiIndex.symmetric_difference` losing extension array (:issue:`48607`)
+- Bug in :meth:`MultiIndex.putmask` losing extension array (:issue:`49830`)
```

Comment:
Would be good to note the performance benefit too in the whatsnew

Reply:
Done